doc/api/async_hooks.md

```diff
@@ -950,6 +950,33 @@ If this method is called outside of an asynchronous context initialized by
 calling `asyncLocalStorage.run` or `asyncLocalStorage.runAndReturn`, it will
 return `undefined`.
 
+### `asyncLocalStorage.enterWith(store)`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `store` {any}
+
+Calling `asyncLocalStorage.enterWith(store)` will transition into the context
+for the remainder of the current synchronous execution and will persist
+through any following asynchronous calls.
+
+Example:
+
+```js
+const store = { id: 1 };
+asyncLocalStorage.enterWith(store);
+asyncLocalStorage.getStore(); // Returns the store object
+someAsyncOperation(() => {
+  asyncLocalStorage.getStore(); // Returns the same object
+});
+```
+
+This transition will continue for the _entire_ synchronous execution.
+This means that if, for example, the context is entered within an event
+handler subsequent event handlers will also run within that context unless
+specifically bound to another context with an `AsyncResource`.
+
```

Comment:
Can you also add a part about error management (as we have for the `run*` methods)

Reply:
Not sure what that would say. This method doesn't _have_ any encapsulating function so the notes on the `run*` methods don't really make sense here. :thinking:

Reply:
I would highlight that. Maybe "as this method does not provide any encapsulation, errors thrown after the call to `enterWith` in a synchronous codepath will all be associated with the context" 🤔

Reply:
Well, they're not really "associated" with the context in any way though. They're not marked or handled in any way by the context. Errors are literally the same if you were _not_ entered into the context. :man_shrugging:

Reply:
By "associated to a context", I mean, what is the return value of `getStore()` when the error handling happens. I agree it's mostly straightforward based on how this method works.

tbh, here I am mostly concerned by domain-replacement cases.

Reply:
Well, that's kind of a separate problem. Async local storage itself shouldn't ever need do anything _itself_ for tracking/handling errors. Domains would just make an instance of `AsyncLocalStorage` and stuff any errors it encounters inside it. By moving the _context_ tracking to `AsyncLocalStorage` however, the need for a _global_ reference should disappear.

Reply:
Fair enough, it will always be time to update the doc should we need it here anyway.